semgrep/tests/e2e/snapshots/test_message_interpolation/test_message_interpolation/rulesmessage_interpolationmulti-pattern-inside.yaml-message_interpolationmulti_pattern_inside_nested.py/results.json

```diff
@@ -10,7 +10,7 @@
       "extra": {
         "is_ignored": false,
         "lines": "            print(\"blah\")",
-        "message": "'print(\"blah\")' detected in method 'B' in class 'A'",
+        "message": "'print(\"blah\")' detected in method 'C' in class 'A'",
```

Comment:
Looking at this rule, this message feels "more correct" to me, but that's open to interpretation :man_shrugging:

Reply:
Some additional context: the `stabilize_evaluation_ordering` function above was created to address ordering, and thus evaluation, differences in different Python versions (Python 3.7+ orders dicts/sets internally by default now, which affects our evaluation). We can flip this back by setting `reverse=False`, however, I think tighter bounds on which code block we're inside (function, class, etc) will provide a better UX. It's easier to grok a finding referring to the innermost block vs. the outermost block which could be many lines away.